pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,639 @@
+func waitForVirtualMachineExport(client kubecli.KubevirtClient, namespace string, interval, timeout time.Duration) error {
+	err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		vmexport, err := getVirtualMachineExport(client, namespace)
+		if err != nil {
+			return false, err
+		}
+
+		if vmexport == nil || vmexport.Status == nil {
+			return false, nil
+		}
+
+		done := vmexport.Status.Phase == "Ready"
+		if !done {
+			fmt.Printf("Waiting for VM Export %s status to be ready...\n", vmexportName)
+			return false, nil
+		}
+
+		if vmexport.Status.Links == nil || vmexport.Status.Links.External == nil {
+			fmt.Printf("Waiting for VM Export %s external links to be available...\n", vmexportName)
+			return false, nil
+		}
+
+		fmt.Printf("Processing completed successfully\n")
+		return done, nil
+	})
+
+	return err
+}
+
+// handleHTTPRequest generates the GET request with proper certificate handling
+func handleHTTPRequest(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, downloadUrl, httpHeader string) (*http.Response, error) {
+	token, err := getTokenFromSecret(client, vmexport.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create new certPool and append our external SSL certificate
+	cert := vmexport.Status.Links.External.Cert
+	roots := x509.NewCertPool()
+	roots.AppendCertsFromPEM([]byte(cert))
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{RootCAs: roots},
+	}
+	httpClient := httpClientCreatorFunc(transport)
+
+	// Generate and do the request
+	req, _ := http.NewRequest("GET", downloadUrl, nil)
+	req.Header.Set(httpHeader, token)
+
+	return httpClient.Do(req)
+}
+
+// getHTTPClient assigns the default, non-mocked HTTP client
+func getHTTPClient(transport *http.Transport) *http.Client {
+	if transport == nil {
+		transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+	}
+
+	client := &http.Client{Transport: transport}
+	return client
+}
+
+// copyFileWithProgressBar serves as a wrapper to copy the file with a progress bar
+func copyFileWithProgressBar(output *os.File, resp *http.Response) error {
+	// TODO: Unable to get file size when downloading using the compressed URL
+	filesize, _ := strconv.Atoi(resp.Header.Get("Content-Length"))
+
+	// Initiate progress bar
+	bar := pb.New(filesize).SetUnits(pb.U_BYTES)
+	rd := bar.NewProxyReader(resp.Body)
+	fmt.Println("Downloading file...")
+	bar.Start()
+
+	_, err := io.Copy(output, rd)
+	bar.Finish()
+	return err
+}
+
+// getOrCreateTokenSecret obtains a token secret to be used along with the virtualMachineExport
+func getOrCreateTokenSecret(client kubecli.KubevirtClient, namespace string) (*k8sv1.Secret, error) {
+	// Generate a random, 20 char string to be used as a token
+	token := rand.String(20)
+
+	secret := &k8sv1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getExportSecretName(),
+			Namespace: namespace,
```

Comment:
Can you add an ownerRef which is the VMExport, this way when the VM export is deleted the secret is also removed.